sceptre/connection_manager.py

```diff
@@ -92,26 +92,27 @@ class ConnectionManager(object):
     _clients = {}
     _stack_keys = {}
 
-    def __init__(self, region, profile=None, stack_name=None, iam_role=None):
+    def __init__(self, region, profile=None, stack_name=None, iam_role=None, duration_seconds=3600):
         self.logger = logging.getLogger(__name__)
 
         self.region = region
         self.profile = profile
         self.stack_name = stack_name
         self.iam_role = iam_role
+        self.duration_seconds = duration_seconds if duration_seconds else 3600
 
         if stack_name:
-            self._stack_keys[stack_name] = (region, profile, iam_role)
+            self._stack_keys[stack_name] = (region, profile, iam_role, duration_seconds)
 
     def __repr__(self):
         return (
             "sceptre.connection_manager.ConnectionManager(region='{0}', "
-            "profile='{1}', stack_name='{2}', iam_role='{3}')".format(
-                self.region, self.profile, self.stack_name, self.iam_role
-            )
+            "profile='{1}', stack_name='{2}', iam_role='{3}', duration_seconds='{4}')".format(
+                self.region, self.profile, self.stack_name, self.iam_role, self.duration_seconds
+             )
         )
 
-    def _get_session(self, profile, region, iam_role):
+    def _get_session(self, profile, region, iam_role, duration_seconds):
```

Comment:
Same as before, I think it duration_seconds shouldn't be a parameter here; Just use the instance variable.